app/util/ScalableRectangle.ts

```diff
@@ -295,14 +300,23 @@ export class ScalableRectangle implements IScalableRectangle {
    * Centers this rectangle on the provided rectangle
    * without changing the scale.
    */
-  centerOn(rect: ScalableRectangle) {
+  centerOn(rect: ScalableRectangle, axis?: CenteringAxis) {
     // Normalize both rectangles for this operation
     this.normalized(() => rect.normalized(() => {
-      // Anchor both rectangles in the center
+      // Anchor both rectangles in the axis center
       this.withAnchor(AnchorPoint.Center, () => {
         rect.withAnchor(AnchorPoint.Center, () => {
-          this.x = rect.x;
-          this.y = rect.y;
+          switch (axis) {
+            case CenteringAxis.X:
+              this.x = rect.x;
+              break;
+            case CenteringAxis.Y:
+              this.y = rect.y;
+              break;
+            default:
+              this.x = rect.x;
+              this.y = rect.y;
+          }
```

Comment:
If you wanted to get fancy here, you could use Typescripts support for bit flags with enums.  You would define your enum like this:

```TypeScript
enum ECenteringAxis {
  X = 1,
  Y = 1 << 1,
  Both = X | Y
}
```
Then you could use it like this:

```TypeScript
if (axis & ECenteringAxis.X) this.x = rect.x;
if (axis & ECenteringAxis.Y) this.y = rect.y;
```

The advantage here is that you don't have to separately handle the "both" case.  Both of those if statements will evaluate to true when the "both" enum is used.

Reply:
Ah got it, makes sense.